## Q1: A先生想要5年後存到200萬元，當作房屋的頭期款，請問每月月底要存入多少錢進土地銀行？假設依據土地銀行一年期定存利率1.36% 計算之。


In [1]:
#導入package
import numpy as np

In [2]:
np.pmt(0.0136/12,12*5,0,-2000000,0)

32231.721506336493

ANS : 32231.72 元

## Q2:A太太為兒子準備大學教育基金，現在離王小弟上大學還有10 年，目前上大學的總花費為80萬元，每年漲幅2%。請問3 年後王先生應準備多少教育基金，投資於年報酬率5%之金融商品，才可供王小弟就讀大學無虞？


In [4]:
## 計算王小弟10年後上大學所需費用
Cost = np.fv(0.02,10,0,-800000)
Cost

975195.5359958059

In [5]:
## 計算王先生3年後需準備多少錢
np.pv(0.05,7,0,-Cost)

693053.2606584559

ANS : 693053.26 元

## Q3:A先生現有300萬元放於投資基金，基金年報酬率5%，預計15年後購置2,000萬元房屋乙幢，則每年平均儲蓄最少應達幾萬元才足夠購屋(假設儲蓄利率3％)？



In [8]:
## 計算300萬放15年得 FV
fund = np.fv(0.05,15,0,-3000000)
fund

6236784.538234106

In [11]:
## 計算差額
gap = 20000000-fund
gap

13763215.461765893

In [12]:
## 計算年付金額
np.pmt(0.03,15,0,-gap)

740001.0315448421

ANS : 每年約存74萬元

## Q4:A先生看上一間40坪的房子，屋主可以出租也可以出售。若租屋每月月初5萬元租金，不用押金。若是購屋，房屋總價6百萬，可以申請到3百萬的房貸，房貸利率為3%，自備款3百萬。假設一年期的定存利率是2%。老王租屋購屋的年成本為何?


In [15]:
## 租屋成本
np.fv(0.02/12,12,-50000,0,1)

606539.8882291287

In [19]:
#計算購屋年成本 計算 頭期款FV+貸款FV - 房屋總價 (月計息,月初付) 
Cost_buy = np.fv(0.02/12,12,0,-3000000,1)+ np.fv(0.03/12,12,0,-3000000) - 6000000
Cost_buy

151800.9377850257

租屋年成本 : 606539.89
購屋年成本 : 151800.93

## Q5:A先生向銀行辦理一筆年利率4%、期間15年之房貸400萬元，約定以本利平均攤還方式按年定期還款，其每年工作結餘50萬元，用以償還房貸並逐年將餘額全數投資至年投資報酬率為3%之基金，則小趙於15年後完成該筆房貸之清償時，另積蓄多少錢？


In [21]:
##計算小趙(A先生)年需繳多少貸款
loan = np.pmt(0.04,15,4000000,0,0)
loan

-359764.4014838926

In [23]:
save = 500000+loan
save

140235.59851610742

In [24]:
np.fv(0.03,15,-save,0,0)

2608229.820649818

ANS : 2608229.82 元